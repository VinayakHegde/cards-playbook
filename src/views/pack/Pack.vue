<script>
import { mapState, mapActions } from "vuex";
import Table from "@/components/table/Table.vue";
import Analysis from "@/components/analysis/Analysis.vue";
export default {
  components: {
    Table,
    Analysis
  },
  props: ["hash"],
  computed: mapState({
    localHash: state => state.hash || ""
  }),
  created: async function() {
    if (!this.localHash) {
      const result = await this.getPile(this.hash);
      let error = false;
      if (result) {
        error = result.error;
      }
      if (error) {
        this.$router.push({ name: "Error" });
      }
    }
  },
  methods: {
    ...mapActions({
      getPile: "getAction"
    })
  }
};
</script>
<template>
  <main>
    <Table />
    <Analysis />
  </main>
</template>
