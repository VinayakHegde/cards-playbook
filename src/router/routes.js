export default [
  {
    path: "/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/new",
    name: "Home",
    meta: {
      header: "CARDS"
    },
    component: () =>
      import(/* webpackChunkName: "home" */ "../views/home/Home.vue")
  },
  {
    path: "/deck/:hash",
    name: "Pack",
    meta: {
      header: "Ordered Pile"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/pack/Pack.vue"),
    props: true
  },
  {
    path: "/error",
    name: "Error",
    meta: {
      header: "404 | Page not found"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/error/Error.vue"),
    props: true
  },
  {
    path: "*",
    redirect: {
      name: "Error"
    }
  }
];
