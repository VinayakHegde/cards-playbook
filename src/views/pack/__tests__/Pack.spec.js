import Vuex from "vuex";
import VueRouter from "vue-router";
import { mount, createLocalVue } from "@vue/test-utils";
import storeConfig from "@/store/store-config";
import routes from "@/router/routes";
import Pack from "../Pack.vue";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
describe("Pack.vue", () => {
  it("match the snapshot", () => {
    const actions = { getAction: jest.fn() };
    const store = new Vuex.Store({ ...storeConfig, actions });
    const router = new VueRouter({ routes });
    const wrapper = mount(Pack, {
      localVue,
      router,
      store,
      propsData: {
        hash: 123
      }
    });
    expect(wrapper).toMatchSnapshot();
    expect(actions.getAction).toHaveBeenCalled();
  });
});
