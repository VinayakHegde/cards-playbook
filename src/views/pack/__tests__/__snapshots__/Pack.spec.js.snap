// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pack.vue match the snapshot 1`] = `
<main>
  <!---->
  <!---->
</main>
`;
